### ISSUE TITLE
Search Failed

### ISSUE THREAD

**Issue by lampholder** [vector-im/riot-web#4603]
There are two elements to this:
1. the search failed
1. the error message for the search's having failed is horrible

The associated rageshake is linked with this issue.

**Comment by turt2live** [comment]
Although rare, I also ran into this while speed-searching for a hitlist of words in a room. Pressing Escape after submitting a search guarantees a CORS error dialog.

**Comment by MadLittleMods** [comment]
Just reproduced this except I don't think I canceled the search at all. Just searched, it was taking a while so I tabbed out and then a while later I came back to this error:

> Search failed:
>
> request failed: CORS request rejected: https://matrix-client.matrix.org/_matrix/client/r0/search

<img width="796" alt="" src="https://user-images.githubusercontent.com/558581/156295269-7b23389a-b3ba-4cbd-8db2-914fb5e2cc03.png">

**Comment by t3chguy** [comment]
That CORS rejection is likely matrix.org's /search API timing out.

**Comment by pedramardakani** [comment]
Yes that was it for me! It would be much better if the error was a bit more clear 😵
